### ISSUE TITLE
ROOT output file support breaks name/type specialization

### ISSUE THREAD

**Issue by simonspa** [simonspa/allpix-squared#44]
In gitlab by @simonspa on Jun 16, 2017, 16:21

Consider the following setup:

## Same propagation for all
```
# propagate
[SimplePropagation]
temperature = 293K # temperature of the sensor
charge_per_step = 100 # amount of charges to propagate at once
```

```
|16:19:27.874|  (STATUS) Welcome to AllPix v0.2beta1+50^gbab7bf2
|16:19:27.874|  (STATUS) Initialized PRNG with seed 13199049227006691428
|16:19:28.062|  (STATUS) Loaded 9 modules                       
|16:19:28.294|  (STATUS) Initialized 13 module instantiations  
|16:19:29.304|  (STATUS) Finished run of 1 events
|16:19:29.323|  (STATUS) Finalization completed
|16:19:29.323|  (STATUS) Executed 13 instantiations in 1 seconds, spending 61% of time in slowest instantiation DepositionGeant4
|16:19:29.323|  (STATUS) Average time per event: 1261 ms
```

## Adjust propagation for one detector

Add specialized propagation module for one detector in order to reduce number of charges carried together:
```
# propagate
[SimplePropagation]
temperature = 293K # temperature of the sensor
charge_per_step = 100 # amount of charges to propagate at once

[SimplePropagation]
name = "detector1"
temperature = 293K # temperature of the sensor
charge_per_step = 10 # amount of charges to propagate at once
```

```
|16:20:00.998|  (STATUS) Welcome to AllPix v0.2beta1+50^gbab7bf2
|16:20:00.998|  (STATUS) Initialized PRNG with seed 12754826517057176659
|16:20:01.195|  (STATUS) Loading module SimplePropagation Error in <TDirectoryFile::mkdir>: An object with name detector1 exists already

|16:20:01.195|   (FATAL) Error during execution of run:
                          Cannot create or access unique ROOT directory for module SimplePropagation:detector1
                         Please check your configuration and modules! Cannot continue...
```

**Comment by simonspa** [comment]
In gitlab by @simonspa on Jun 16, 2017, 16:21

To add,

 1. Order does not play a role
 2. order should not play a role

**Comment by simonspa** [comment]
In gitlab by @Koensw on Jun 16, 2017, 17:16

Confirmed, this is wrong. Currently it instantiates the directories during construction.

This should either move to before the initialization (when the specializations are sorted out) or it should take over the directory created earlier.

Anyway the directories should not be accessed in the constructor (these things should also be clearer in the documentation)

**Comment by simonspa** [comment]
In gitlab by @Koensw on Jun 19, 2017, 11:43

closed via commit e48ddaa015487c960fb61ed7412cbadca109927a
